File: CS/DXSample/MainViewModel.cs
using DevExpress.Xpf.Mvvm;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Data;
using System.Windows.Documents;
using System.Windows.Input;
using System.Windows.Media;
using System.Windows.Media.Imaging;
using System.Windows.Navigation;
using System.Windows.Shapes;

namespace DXSample {
    public class MainViewModel : ViewModelBase {
        public ICommand OnViewLoadedCommand { get; private set; }
        public ICommand GoToUserControl1Command { get; private set; }
        public ICommand GoToUserControl2Command { get; private set; }
        public ICommand GoToUserControl3Command { get; private set; }
        public MainViewModel() {
            OnViewLoadedCommand = new DelegateCommand(OnNavigateDetailsCommandExecute);
            GoToUserControl1Command = new DelegateCommand(GoToUserControl1CommandExecute);
            GoToUserControl2Command = new DelegateCommand(GoToUserControl2CommandExecute);
            GoToUserControl3Command = new DelegateCommand(GoToUserControl3CommandExecute);

        }
        void OnNavigateDetailsCommandExecute() {
            //Navigate to the HomeView.
            ServiceContainer.GetService<INavigationService>().Navigate("HomeView", null, this);
        }
        private void GoToUserControl1CommandExecute() {
            ServiceContainer.GetService<INavigationService>().Navigate("UserControl1", null, this);
        }
        private void GoToUserControl2CommandExecute() {
            ServiceContainer.GetService<INavigationService>().Navigate("UserControl2", null, this);
        }
        private void GoToUserControl3CommandExecute() {
            ServiceContainer.GetService<INavigationService>().Navigate("UserControl3", null, this);
        }
    }
}

File: VB/DXSample/MainViewModel.vb
Imports DevExpress.Xpf.Mvvm
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Windows
Imports System.Windows.Controls
Imports System.Windows.Data
Imports System.Windows.Documents
Imports System.Windows.Input
Imports System.Windows.Media
Imports System.Windows.Media.Imaging
Imports System.Windows.Navigation
Imports System.Windows.Shapes

Namespace DXSample
	Public Class MainViewModel
		Inherits ViewModelBase

		Private privateOnViewLoadedCommand As ICommand
		Public Property OnViewLoadedCommand() As ICommand
			Get
				Return privateOnViewLoadedCommand
			End Get
			Private Set(ByVal value As ICommand)
				privateOnViewLoadedCommand = value
			End Set
		End Property
		Private privateGoToUserControl1Command As ICommand
		Public Property GoToUserControl1Command() As ICommand
			Get
				Return privateGoToUserControl1Command
			End Get
			Private Set(ByVal value As ICommand)
				privateGoToUserControl1Command = value
			End Set
		End Property
		Private privateGoToUserControl2Command As ICommand
		Public Property GoToUserControl2Command() As ICommand
			Get
				Return privateGoToUserControl2Command
			End Get
			Private Set(ByVal value As ICommand)
				privateGoToUserControl2Command = value
			End Set
		End Property
		Private privateGoToUserControl3Command As ICommand
		Public Property GoToUserControl3Command() As ICommand
			Get
				Return privateGoToUserControl3Command
			End Get
			Private Set(ByVal value As ICommand)
				privateGoToUserControl3Command = value
			End Set
		End Property
		Public Sub New()
			OnViewLoadedCommand = New DelegateCommand(AddressOf OnNavigateDetailsCommandExecute)
			GoToUserControl1Command = New DelegateCommand(AddressOf GoToUserControl1CommandExecute)
			GoToUserControl2Command = New DelegateCommand(AddressOf GoToUserControl2CommandExecute)
			GoToUserControl3Command = New DelegateCommand(AddressOf GoToUserControl3CommandExecute)

		End Sub
		Private Sub OnNavigateDetailsCommandExecute()
			'Navigate to the HomeView.
			ServiceContainer.GetService(Of INavigationService)().Navigate("HomeView", Nothing, Me)
		End Sub
		Private Sub GoToUserControl1CommandExecute()
			ServiceContainer.GetService(Of INavigationService)().Navigate("UserControl1", Nothing, Me)
		End Sub
		Private Sub GoToUserControl2CommandExecute()
			ServiceContainer.GetService(Of INavigationService)().Navigate("UserControl2", Nothing, Me)
		End Sub
		Private Sub GoToUserControl3CommandExecute()
			ServiceContainer.GetService(Of INavigationService)().Navigate("UserControl3", Nothing, Me)
		End Sub
	End Class
End Namespace
